sys/include/can/can.h

```diff
@@ -95,6 +95,86 @@ struct can_frame {
     uint8_t data[CAN_MAX_DLEN] __attribute__((aligned(8)));
 };
 
+#if defined(MODULE_CAN_FILTER_TYPE_SPECIFIC)
+/**
+ * @brief Controller Area Network filter type
+ *
+ * The different CAN controllers provide different CAN filter
+ * types. These types are provided by the SAMD5x family (please check
+ * chapter 39.9.5 and 39.9.6 of the SAMD5x/SAME5x family Datasheet).
+ *
+ * The current CAN filter types list is from the SAMD5x/SAME5x family.
+ *
+ * Can be extended by other filter types if provided
+ */
+typedef enum {
+    CAN_FILTER_TYPE_CLASSIC = 0x00,     /**< Classic Filter: Filter ID and Mask */
+    CAN_FILTER_TYPE_RANGE,              /**< Range filter from Filter 1 to Filter 2 (Filter 2 > Filter 1) */
+    CAN_FILTER_TYPE_DUAL,               /**< Dual ID Filter (Filter 2 or Filter 1) */
+    CAN_FILTER_TYPE_EXT_RANGE           /**< For extended filters, Range filter from Filter 1 to Filter 2 (Filter 2 > Filter 1) */
+} can_filter_type_t;
+
+/**
+ * @brief Controller Area Network filter configuration
+ *
+ * The different CAN controllers provide different CAN filter
+ * configurations. These types are provided by the SAMD5x family (please check
+ * chapter 39.9.5 and 39.9.6 of the SAMD5x/SAME5x family Datasheet).
+ *
+ * The current CAN filter configuration list is from the SAMD5x/SAME5x family.
+ *
+ * Can be extended by other filter configurations if provided
+ */
+typedef enum {
+    CAN_FILTER_DISABLE = 0x00,      /**< Disable Filter element */
+    CAN_FILTER_RX_0,                /**< Store message in reception mailbox Rx_0 if filter matches */
+    CAN_FILTER_RX_1,                /**< Store message in reception mailbox Rx_1 if filter matches */
+    CAN_FILTER_RX_REJECT,           /**< Reject message if filter matches */
+    CAN_FILTER_RX_PRIO,             /**< Set priority if filter matches */
+    CAN_FILTER_RX_PRIO_0,           /**< Set priority and store message in reception mailbox Rx_0 if filter matches */
+    CAN_FILTER_RX_PRIO_1,           /**< Set priority and store message in reception mailbox Rx_1 if filter matches */
+    CAN_FILTER_RX_STRXBUF           /**< Store message in the RX buffer or as debug message */
+} can_filter_conf_t;
```

Comment:
This is still very much SAM E54 specific - why not move the target mailbox to a separate member (as it appears to be a common thing across CAN controllers) and only have generic  toggles in the flags struct?

Reply:
I'm not sure why disable filter needs to be a flag - if you want to disable the filter, just do so via an appropriate API instead of 'adding' a filter with the disable flag set.

I assume the 'REJECT' filter is to make a more generic 'ACCEPT' filter more fine grained?
I think we don't need an 'ACCEPT' flag as 'ACCEPT' is the default (we reject all CAN frames for which we don't have a filter), so we'd only need a flag if we want to 'invert' the filter and also reject frames that match it.

Reply:
I've updated it accordingly